src/ipc/handler.c

```diff
@@ -104,17 +104,37 @@
 
 #define _IPC_COPY_CMD(rx, tx, rx_size)					\
 	do {								\
+		int err = 0;						\
 		if (rx_size > tx->size) {				\
-			memcpy(rx, tx, tx->size);			\
+			err = memcpy_s(rx, rx->size, tx, tx->size);	\
```

Comment:
`rx->size` may be uninitialized in many calls. `rx_size` should be used instead.